samples/apps/json.jakt

```diff
@@ -223,37 +245,95 @@ class JsonParser {
         return JsonValue::Object(values)
     }
 
+    function char_to_f64(anon num: u8) throws -> f64 {
+        // FIXME 1: Shouldn't need this function at all
+        // FIXME 2: Shouldn't need return in else branch
+        return match num {
+            0u8 => 0.0
+            1u8 => 1.0
+            2u8 => 2.0
+            3u8 => 3.0
+            4u8 => 4.0
+            5u8 => 5.0
+            6u8 => 6.0
+            7u8 => 7.0
+            8u8 => 8.0
+            9u8 => 9.0
+            else => {
+                // FIXME: "Unexpected number"
+                throw Error::from_errno(9017)
+                return 1.0
+            }
+        }
+    }
+
     function parse_number(mut this) throws -> JsonValue {
+        // FIXME: This implementation doesn't match JsonParser.cpp
         let is_negative = .consume_specific(b'-')
+        mut decimal_start_index: usize? = None
 
-        mut value = 0
+        mut value = 0.0
 
         while not .eof() {
             let ch = .peek()
-            if not (ch >= b'0' and ch <= b'9') {
+            if ch == b'.' {
+                if decimal_start_index.has_value() {
+                    // FIXME: "Unexpected '.'"
+                    throw Error::from_errno(9016)
+                }
+                decimal_start_index = .index++
+                continue
+            } else if not (ch >= b'0' and ch <= b'9') {
                 break
             }
-            value *= 10
-            value += (ch - b'0') as! i64
+
+            if not decimal_start_index.has_value() {
+                value *= 10.0
+                value += char_to_f64(ch - b'0')
+            } else {
+                mut num = char_to_f64(ch - b'0')
+                // FIXME: This should really be: `value += pow(10, -decimal_place)*num`, but: there's no pow function and you can't multiply float by usize
```

Comment:
FIXME: use pow function when implemented

Reply:
If you can't cast usize to f64 then add another fixme `FIXME: can't cast to float from usize`